lib/ansiblelint/errors.py

```diff
@@ -27,3 +45,24 @@ def __repr__(self):
 
         return formatstr.format(_id, self.message,
                                 self.filename, self.linenumber, self.line)
+
+    @property
+    def _hash_key(self):
+        # line attr is knowingly excluded, as dict is not hashable
```

Comment:
Have you considered turning dict into a tuple of tuples for the hashing purposes?

Reply:
Not really because it would be outside the current change scope. I do not want to rewrite half of the ansible-lint to make this change happen.

I am not against such idea, but I know for sure that the only way to progress is in small steps, not big rewrites that never end being merged.

The scope of a review is to assure that now new bugs are introduced and that the overall code quality is improved, not that is "perfect" or that it also fixes any other per-existent issue.

Shortly, I would try to contain changes and minimize them. We should raise bugs for found issues, maybe even assign them.

Reply:
I totally support the idea of small incremental changes. The reason I'm raising this question is that I suspect that in the current state it'll negatively influence the filtering logic. That is when there's two different matches, it'll randomly remove one of them as a duplicate. Right? I'd consider that a bug.

Reply:
@ssbarnea ^

Reply:
I removed the set() part, so we no longer remove duplicates. We can analyze and make a decision in another change. The best is to do this when we have at least one example or two of duplicated results.